client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -131,10 +133,6 @@ const NewMailBoxList = (
 	}, [ mailboxes ] );
 
 	useEffect( () => {
-		if ( hiddenFieldNames.length < 1 ) {
-			return;
-		}
-
```

Comment:
Looks like we might need to add this back to get unit tests to pass. Alternatively (which I think might actually be a better solution), we change the dependency array to something like `[ ...hiddenFieldNames ]` instead of `[ hiddenFieldNames ]`.

Reply:
Good point!

Reply:
Even if the unit tests pass now, I think it remains a bit risky to not have this, because passing a non-stable `hiddenFieldNames` array is going to trigger an infinite re-render cycle.

Reply:
That line was added to fix the test, seems to me as a workaround. It worked while TITAN had 2 hidden fields always (Name and Alternative Email). Now that Google has only 1, it never disappears.

Reply:
To be more accurate, the infinite re-render cycle only happened in tests, exactly because of [this](https://github.com/Automattic/wp-calypso/pull/67420/files#diff-0b00cb0ebb76dd060d060350053de3dffb9f570a6d504a505bfd36935fb18fcbR44).

We are passing an array of dependencies already (hidden fields) so there is no risk of infinite rendering.

Reply:
Sure! But using `[ hiddenFieldNames ]` as a dependency array means that we are passing a reference type in the dependency array. So even if the array members of `hiddenFieldNames` don't change between renders, the hook would still be called if `hiddenFieldNames` is a new array instance. 

I know that you didn't make any change to the dependency array in this PR, but this might be a nice opportunity to make the change. 

Not a blocker since tests are passing and the code works, but still.

Reply:
No expert in react here 😅 But I did run into the issue of infinite re-render when I worked on this fix [here](https://github.com/Automattic/wp-calypso/pull/66373/files) and I didn't understand why. I had to set the default hidden fields a specific way, in order for to avoid infinite re-render. Not sure if that might be relevant to this discussion here.

Reply:
I maintain that we should change the `useEffect` dependency. This solution is tested and works:
```js
}, [ hiddenFieldNames.join( '' ) ] ); // eslint-disable-line react-hooks/exhaustive-deps
```

Reply:
@wongasy the dependency array prevents the code to run if the parameters has not changed. it's used for several reasons, one is to avoid infinite renders, example:

render form -> set visible fields (html tree changed on useEffect) -> render (triggered as consequence of tree changed) ->set visible fields (html tree did not change, but did modify the html adding/removing) -> render (triggered as consequence of html modification, even if it result on same HTML) -> infinite loop.

So what array dependencies does is to provide parameters, to compare the previous execution with the current execution, as function should be deterministic (same parameters produce same result) the useEffect won't be triggered as the function did not change its parameters, and will use the same effect as before.

Another reason is to avoid expensive calculations (either CPU extensive, or ones that involves HTTP requests).

@fredrikekelund I'm adding the join